sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -286,6 +286,18 @@ public int hashCode() {
      */
     IterableAssert<T> inEarlyGlobalWindowPanes();
 
+    /**
+     * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
+     * run on additions.
+     */
+    IterableAssert<T> filterAdditions();
```

Comment:
The most important test is to `resolveRetractions` to get the true contents. The javadoc for these two should tell users to use that. Because triggers are nondeterministic and the really important thing is to test that whatever comes out is equivalent. And only special tests to make sure triggers are firing need to have these other tests. They basically aren't really for most users.